### ISSUE TITLE
Collection description is required on the frontend but in the api is optional

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-frontend#11052]
### Describe the problem and steps to reproduce it:

1. Log in on AMO stage
2. Create a new collection: fill in only the collection name and the slug (should be automatically generated from the name input)
3. Save the collection

### What happened?
"This field may not be blank" error received. The error is caused by the description field.

### What did you expect to happen?
The description field hasn't been mandatory until now and It wasn't clear what made it so. Also, the field is optional in the collections create API.

![image](https://user-images.githubusercontent.com/31961530/145175865-5fdc6fd8-ad2d-4b8a-9bb8-d6a26e6498ed.png)

**Comment by bobsilverberg** [comment]
When verifying that this behaviour doesn't exist on prod, I came across another weird validation failure when creating a collection:

![Screen Shot 2021-12-08 at 2 56 28 PM](https://user-images.githubusercontent.com/142755/145275442-3b420d5c-1330-46c7-9bbf-e3c568c018e0.png)

Looking at the data sent to the API in the network panel, I see `"slug":"A-new-collection"`, which seems entirely valid to me.

@eviljeff Do you know if anything landed recently that could have affected the behaviour of the "create a collection" API?

**Comment by AlexandraMoga** [comment]
There's something interesting going on with the collection naming and the slug formation:

This is valid:
![image](https://user-images.githubusercontent.com/31961530/145357770-f5789f5a-2386-47d5-b59a-bb450abaf04f.png)

This is not valid:
![image](https://user-images.githubusercontent.com/31961530/145357917-be7ba93f-335a-4206-a0eb-f77898c6e3a4.png)


So, it seems the field cannot handle uppercase letters. This is definitely an API issue, but the initial report about the Description field I'm not so sure, since I can create a collection directly from the API without providing a description.

Also, I was able to find a collection with upper letters in the slug among my collections, which means the issue didn't exist at some point: https://addons-dev.allizom.org/en-US/firefox/collections/10642146/Collection-of-December-2021/. As expected, I can't edit the collection details for this example anymore as I'm running into the error mentioned above.

**Comment by eviljeff** [comment]
@bobsilverberg you're talking about a different bug - can you file it separately.

**Comment by eviljeff** [comment]
@AlexandraMoga it's likely a regression caused by other API work (e.g #18200) - there's a lot of common classes if there's no test failing it's difficult to know.  It should be optional.

**Comment by eviljeff** [comment]
@bobsilverberg are you wanting to take this on?  It's almost certainly API related.

**Comment by eviljeff** [comment]
Though now I think some more on this, is it possible that frontend is always sending `description` as an empty string (or rather `{en-US: ''}`)?  The field is `required=False` in the API but if _something_ is sent - and an object is something, even if the values are empty strings, then it'll be validated.  

If so, there are two fixes - either change `CollectionSerializer` in addons-server to allow empty strings; or change addons-frontend to not send objects with empty string values in it.  (tbh, the latter seems like the "correct" fix to me)

**Comment by AlexandraMoga** [comment]
Looking at the request sent by the frontend when the form is submitted, looks like description is indeed sent as an empty string:

```
{
	"default_locale": "en-US",
	"description": {
		"en-US": ""
	},
	"name": {
		"en-US": "no description"
	},
	"slug": "no-description"
}
```

**Comment by bobsilverberg** [comment]
> @bobsilverberg are you wanting to take this on? It's almost certainly API related.

I can take it on, but was hoping for some insight from you if you have some ideas about what changes that landed might be affecting it. I see you added a comment at https://github.com/mozilla/addons-frontend/issues/11052#issuecomment-989902663 about front-end passing an empty string, along with a suggestion for a fix. The front-end behaviour has not changed, so I gather that something changed in the API code that is now causing this extra validation?

**Comment by eviljeff** [comment]
> > @bobsilverberg are you wanting to take this on? It's almost certainly API related.
> 
> I can take it on, but was hoping for some insight from you if you have some ideas about what changes that landed might be affecting it. I see you added a comment at [#11052 (comment)](https://github.com/mozilla/addons-frontend/issues/11052#issuecomment-989902663) about front-end passing an empty string, along with a suggestion for a fix. The front-end behaviour has not changed, so I gather that something changed in the API code that is now causing this extra validation?

it's mozilla/addons-server#18377- it fixed a (long standing) bug where you could submit empty strings in an object for a required field and the value be regarded as valid.

**Comment by bobsilverberg** [comment]
> Though now I think some more on this, is it possible that frontend is always sending `description` as an empty string (or rather `{en-US: ''}`)? The field is `required=False` in the API but if _something_ is sent - and an object is something, even if the values are empty strings, then it'll be validated.
> 
> If so, there are two fixes - either change `CollectionSerializer` in addons-server to allow empty strings; or change addons-frontend to not send objects with empty string values in it. (tbh, the latter seems like the "correct" fix to me)

@eviljeff This makes sense, but what happens in the case that there is currently a `description` and the user wishes to remove it? If addons-frontend sends nothing if `description` is blank, then it would be impossible to change an existing `description` to be blank. It seems to me that allowing a collection to be created with an empty `description` implies that `description` itself is not a required field, so shouldn't the rules on addons-server be changed so that it is not treated like a required field, and addons-frontend should continue to send an empty string in the API request?

**Comment by eviljeff** [comment]
> This makes sense, but what happens in the case that there is currently a `description` and the user wishes to remove it? If addons-frontend sends nothing if `description` is blank, then it would be impossible to change an existing `description` to be blank. 

yeah... mozilla/addons-server#8816 is unfortunately still an open issue.  So I guess we'll have to keep allowing setting an empty string as alternative to actually being able to delete it.

> It seems to me that allowing a collection to be created with an empty `description` implies that `description` itself is not a required field, so shouldn't the rules on addons-server be changed so that it is not treated like a required field, and addons-frontend should continue to send an empty string in the API request?

It's not a required field, or treated like a required field.  The error in the screenshot is that it can't be _blank_, i.e. an empty string, or more specifically in this case, that the localized values can't be empty strings.  The default is `allow_blank=False`, but it can changed in the serializer.

**Comment by bobsilverberg** [comment]
> 
> It's not a required field, or treated like a required field. The error in the screenshot is that it can't be _blank_, i.e. an empty string, or more specifically in this case, that the localized values can't be empty strings. The default is `allow_blank=False`, but it can changed in the serializer.

Ok, so this should be transferred to addons-server and fixed in the serializer?

**Comment by eviljeff** [comment]
> > It's not a required field, or treated like a required field. The error in the screenshot is that it can't be _blank_, i.e. an empty string, or more specifically in this case, that the localized values can't be empty strings. The default is `allow_blank=False`, but it can changed in the serializer.
> 
> Ok, so this should be transferred to addons-server and fixed in the serializer?

probably
